mmv1/third_party/terraform/services/colab/resource_colab_schedule_test.go

```diff
@@ -0,0 +1,539 @@
+package colab_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccColabSchedule_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"location":           envvar.GetTestRegionFromEnv(),
+		"project_id":         envvar.GetTestProjectFromEnv(),
+		"service_account":    envvar.GetTestServiceAccountFromEnv(t),
+		"end_time":           time.Date(time.Now().Year(), 12, 31, 0, 0, 0, 0, time.Now().Location()).AddDate(0, 0, 10).Format(time.RFC3339),
+		"start_time":         time.Date(time.Now().Year(), 12, 31, 0, 0, 0, 0, time.Now().Location()).AddDate(0, 0, 1).Format(time.RFC3339),
+		"random_suffix":      acctest.RandString(t, 10),
+		"updated_start_time": time.Date(time.Now().Year(), 12, 31, 0, 0, 0, 0, time.Now().Location()).AddDate(0, 0, 2).Format(time.RFC3339),
+		"updated_end_time":   time.Date(time.Now().Year(), 12, 31, 0, 0, 0, 0, time.Now().Location()).AddDate(0, 0, 5).Format(time.RFC3339),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckColabScheduleDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccColabSchedule_full(context),
+			},
+			{
+				ResourceName:            "google_colab_schedule.schedule",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location"},
+			},
+			{
+				Config: testAccColabSchedule_update(context),
```

Comment:
As documented at https://googlecloudplatform.github.io/magic-modules/test/test/#add-an-update-test, you can add a check here to ensure that the update step isn't accidentally deleting the resource:

```suggestion
				Config: testAccColabSchedule_update(context),
				ConfigPlanChecks: resource.ConfigPlanChecks{
					PreApply: []plancheck.PlanCheck{
						plancheck.ExpectResourceAction("google_colab_schedule.schedule", plancheck.ResourceActionUpdate),
					},
				},
```